akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -146,27 +182,62 @@ object Behavior {
    */
   def validateAsInitial[T](behavior: Behavior[T]): Behavior[T] =
     behavior match {
-      case `sameBehavior` | `unhandledBehavior` ⇒
+      case SameBehavior | UnhandledBehavior ⇒
         throw new IllegalArgumentException(s"cannot use $behavior as initial behavior")
       case x ⇒ x
     }
 
   /**
-   * Validate the given behavior as initial, pass it a [[PreStart]] message
-   * and canonicalize the result.
+   * Validate the given behavior as initial, initialize it if it is deferred.
    */
   def preStart[T](behavior: Behavior[T], ctx: ActorContext[T]): Behavior[T] = {
-    val b = validateAsInitial(behavior)
-    if (isAlive(b)) canonicalize(b.management(ctx, PreStart), b) else b
+    validateAsInitial(behavior) match {
+      case d: DeferredBehavior[_] ⇒ validateAsInitial(undefer(d.asInstanceOf[DeferredBehavior[T]], ctx))
+      case x                      ⇒ x
+    }
   }
 
   /**
    * Returns true if the given behavior is not stopped.
    */
-  def isAlive[T](behavior: Behavior[T]): Boolean = behavior ne stoppedBehavior
+  def isAlive[T](behavior: Behavior[T]): Boolean = behavior ne StoppedBehavior
 
   /**
    * Returns true if the given behavior is the special `Unhandled` marker.
    */
-  def isUnhandled[T](behavior: Behavior[T]): Boolean = behavior eq unhandledBehavior
+  def isUnhandled[T](behavior: Behavior[T]): Boolean = behavior eq UnhandledBehavior
+
+  /**
+   * Execute the behavior with the given message
+   */
+  def interpretMessage[T](behavior: Behavior[T], ctx: ActorContext[T], msg: T): Behavior[T] =
+    interpret(behavior, ctx, msg)
+
+  /**
+   * Execute the behavior with the given signal
+   */
+  def interpretSignal[T](behavior: Behavior[T], ctx: ActorContext[T], signal: Signal): Behavior[T] =
+    interpret(behavior, ctx, signal)
```

Comment:
These two methods may seem a bit superfluous, but I didn't want to repeat the interpret logic and I didn't want to leak the untyped msg nature of `interpret` to the rest of the code base.